### ISSUE TITLE
Prune removes packs if they contain needed blobs multiple times

### ISSUE THREAD

**Issue by aawsome** [rustic-rs/rustic#392]
The prune check for needed pack didn't work for packs containing identical blobs multiple times. As a result prune marked such a pack for removal instead of keeping or repacking it.

Even worse - the backup command currently does create packs with multiple identical blobs, see https://github.com/rustic-rs/rustic/pull/393 . So this is critical and can cause data loss when using prune!

Steps to reproduce the error:
```
# create a pack with multiple identical blobs
dd if=/dev/zero bs=10M count=1 | rustic  backup - 
# removes the needed pack
rustic prune --instant-delete 
```

Note: The problem of immediate data loss is mitigated by the two-phase pruning which only marks the pack for removal. But the second phase would still finally remove the file..

**Comment by aawsome** [comment]
The prune bug is fixed by #391.
The reason for multiple identical blobs in a pack is fixed by #391.

I leave this issue open until the fixes are released.

**Comment by aawsome** [comment]
I'm closing this issue. The bug is fixed in 0.4.3 - please update to the latest release!
